### PR TITLE
linked domain history report

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

File: corehq/apps/linked_domain/models.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ class DomainLink(models.Model):
     linked_domain = models.CharField(max_length=126, null=False, unique=True)
     master_domain = models.CharField(max_length=126, null=False)
     last_pull = models.DateTimeField(null=True, blank=True)
+    deleted = models.BooleanField(default=False)
```
Comment (esoergel) [inline corehq/apps/linked_domain/models.py]: You may want to [modify the base queryset](https://docs.djangoproject.com/en/1.11/topics/db/managers/#modifying-a-manager-s-initial-queryset) to exclude deleted results unless you do something explicit.  For instance:
```python
DomainLink.objects.all()  # does not include deleted objects
DomainLink.deleted_objects.all()  # only includes deleted objects
```

File: corehq/apps/linked_domain/admin.py [hunk 2]
```diff
@@ -24,6 +25,18 @@ class DomainLinkAdmin(admin.ModelAdmin):
...
+        'delete', 'undelete'
+    ]
+
+    def delete(self, request, queryset):
+        queryset.update(deleted=True)
+    delete.short_description = "Mark selected items as deleted"
+
+    def undelete(self, request, queryset):
+        queryset.update(deleted=False)
+    undelete.short_description = "Undelete selected items"
```
Comment (esoergel) [inline corehq/apps/linked_domain/admin.py]: Cool!

File: corehq/apps/linked_domain/views.py [hunk 3]
```diff
@@ -186,3 +192,114 @@ def delete_domain_link(self, in_data):
...
+        return self.request.GET.get('domain_link_model')
+
+    @property
+    @memoized
+    def domain_link(self):
+        if self.request.GET.get('domain_link'):
+            try:
+                return DomainLink.objects.get(pk=self.request.GET.get('domain_link'))
+            except DomainLink.DoesNotExist:
+                pass
```
Comment (esoergel) [inline corehq/apps/linked_domain/views.py]: Should this 404?

**Review by esoergel — APPROVED** [review]

File: corehq/apps/linked_domain/views.py (hunk 3 above)
Comment (snopoke) [reply corehq/apps/linked_domain/views.py]: Raising a 404 here would 404 the whole report. There is a check in `get_data` that returns no rows if the link doesn't exist.

File: corehq/apps/linked_domain/models.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/linked_domain/models.py]: b961a66

**Comment by snopoke** [comment]
@esoergel Updated

**Review by esoergel — APPROVED** [review]
thanks!

**Pull request merged**